lms/static/scripts/frontend_apps/components/dashboard/DashboardActivityFilters.tsx

```diff
@@ -187,30 +322,18 @@ export default function DashboardActivityFilters({
     Student[],
     StudentsResponse
   >('students', routes.students, studentsFilters);
-  const studentsWithFallbackName: FiltersStudent[] | undefined = useMemo(
-    () =>
-      studentsResult.data?.map(({ display_name, ...s }) => ({
-        ...s,
-        display_name:
-          display_name ??
-          `Student name unavailable (ID: ${s.lms_id.substring(0, 5)})`,
-        has_display_name: !!display_name,
-      })),
-    [studentsResult.data],
-  );
```

Comment:
This is now handled by the `StudentOption` local component, so that we can use APi fetch results consistently for the three filters.